### PR TITLE
REDUX SETUP

### PR DESCRIPTION
**Pull request by maxpinheiro**
the best PR we will have this semester

### CODE REVIEW

File: store/Auth.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    export interface State {
+        authenticated: boolean
+    }
+
+    const initialState: State = {
+        authenticated: false
+    }
+
+    // todo: i think this needs to be a union or something
+    export type Action = ActionType<boolean>;
```
Comment (nicoledanuwidjaja) [inline store/Auth.reducer.ts:15]: yeah we can expand on this

File: store/Auth.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    export const authReducer = (state = initialState, action: AnyAction): State => {
+        switch (action.type) {
+            case AuthAction.Type.SET_AUTHENTICATED:
+                return { ...state, authenticated: Boolean(action.payload) };
```
Comment (nicoledanuwidjaja) [inline store/Auth.reducer.ts:20]: Add the cookies here

File: store/Auth.reducer.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { AnyAction } from "redux";
+import { ActionType } from "./App.store";
+import { AuthAction } from "./Auth.action";
+
+export namespace AuthReducer {
+    export interface State {
+        authenticated: boolean
```
Comment (nicoledanuwidjaja) [inline store/Auth.reducer.ts:7]: If we want to store the cookies add the string here!

**Review by nicoledanuwidjaja — APPROVED** [review]
Woohoo!

**Review by connorlbark — APPROVED** [review]
<3 love ya

**Pull request merged**